### ISSUE TITLE
Customization, General Questions and Speed Up

### ISSUE THREAD

**Issue by yukiarimo** [RVC-Boss/GPT-SoVITS#1729]
Hello. I’m very excited about the big V2 update and have managed to fine-tune your model on 5 hours of my high-quality data. However, I have a few questions:

1. I was using the Colab version here: https://rentry.co/GPT-SoVITS-Colab and followed the tutorial, a bit to modifying the batch size to make it faster. In LLMs, I’m using the HF trainer, which has many training parameters. But where are they in this model? I can only see three parameters.
2. I’ve tried DPO, but after five epochs, the output was horrible, random Chinese garbage (I'm doing everything in English)! Why is this happening? Without DPO, everything was fine-tuned fine (even at these five epochs). Please explain more about DPO and what it does. I understand the concept of DPO in LLMs, like the correct and wrong answers, but I’m not sure about this one.
3. Are there more parameters than top_k, top_p, temperature, and repetition penalty? I know that Transformers (like LLaMA) have more parameters.
4. Is this model convertible to VITS? I’ve seen many raw VITS models on HuggingFace, but this seems different. Also, there’s ONNX stuff that doesn’t work. Please fix it.
5. Also, what batch size should I use for my MacBook Pro M1 with 16GB of RAM? I tried 40, but it doesn’t make much of a difference (device set to MPS). Is training the SoVITS model for 30 epochs and GPT for 40 epochs enough? And why is the learning rate set to 0.5 if in LLMs, it's often like 4e-5?

**Comment by XXXXRT666** [comment]
Do not using mps

**Comment by XXXXRT666** [comment]
Also, there is a prefix for learning rate

**Comment by XXXXRT666** [comment]
And the optimizer is important, you'd better using pytorch-lightning for training.

For DPO, please specify what happened with logs in terminal. 

You can check the source code for parameters, but I believe there is only a few.

It's not VITS, but it's modified from VITS, and you could try torch script.

If you only got few audios, just train a few epochs to avoid overfitting.

After reading other issues, you will know why cpu training and inference for Mac

**Comment by yukiarimo** [comment]
> Do not use mps

I’m training on Google Colab. But should I use MPS or CPU for the inference, then? Yeah, and what about the batch size?

> It's not VITS, but it's modified from VITS, and you could try torch script.

Can you please, in simple terms, tell me what's changed? (Just out of the curiosity)

> And the optimizer is important, you'd better using pytorch-lightning for training.

Oh, I'm not sure how to do it with the PyTorch Lighting. Do you already have the code for that? Also, how do I enable the tensorboard or just any logging, since there are no logs in the Colab?

> Also, there is a prefix for learning rate

What prefix?

**Comment by XXXXRT666** [comment]
> I’m training on Google Colab. But should I use MPS or CPU for the inference, then? Yeah, and what about the batch size?
>
Do not use MPS in GPT part for any usage
> Can you please, in simple terms, tell me what's changed? (Just out of the curiosity)
> 
Using HuBert to enhance emotion transfer and zero-shot similarity
> Oh, I'm not sure how to do it with the PyTorch Lighting. Do you already have the code for that? Also, how do I enable the tensorboard or just any logging, since there are no logs in the Colab?
> 
Just use the training WebUI, and the logs are in `logs`, you can view the line charts using tensor board directly, but I don't think it will be useful since it's just a few steps fine-tune
> What prefix?
> 
Text model weight learning rate, not the GPT model

**Comment by Lion-Wu** [comment]
Use CPU, and ensure `is_half` is set to `false`(set by default) for best training and inference experience on Mac.
There can be issues with MPS now.

> I’m training on Google Colab. But should I use MPS or CPU for the inference, then?

You can use a CPU to perform inference on a model trained with a GPU.

To adjust batch size, divide available memory (GB) by 2, but you can sometimes go higher, just avoid exceeding total memory and prevent swap usage.

**Comment by yukiarimo** [comment]
Created custom TTS. No need. Closing.
